src/runtime_src/tools/xclbinutil/Section.cxx

```diff
@@ -31,16 +32,15 @@ namespace XUtil = XclBinUtilities;
   #pragma warning( disable : 4100)      // 4100 - Unreferenced formal parameter
 #endif
 
-
-
-// Static Variables Initialization
-std::map<enum axlf_section_kind, std::string> Section::m_mapIdToName;
-std::map<std::string, enum axlf_section_kind> Section::m_mapNameToId;
-std::map<enum axlf_section_kind, Section::Section_factory> Section::m_mapIdToCtor;
-std::map<std::string, enum axlf_section_kind> Section::m_mapJSONNameToKind;
-std::map<enum axlf_section_kind, std::string> Section::m_mapKindToJSONName;
-std::map<enum axlf_section_kind, bool> Section::m_mapIdToSubSectionSupport;
-std::map<enum axlf_section_kind, bool> Section::m_mapIdToSectionIndexSupport;
+Section::FactoryData& Section::factoryData()
+{
+  // We allocate an object instead of using a local static to avoid potential
+  // order problems during teardown destruction. One might suggest this is a
+  // memory leak, but it will be reclaimed by the OS when the program exits.
+  // See https://isocpp.org/wiki/faq/ctors#construct-on-first-use-v2
+  static FactoryData* pData = new FactoryData();
+  return *pData;
```

Comment:
Not following what the problem might be.  If some static global object some place is referencing the FactoryData object then that would be bad and wrong.   By allocating on the heap, proper correct use of the 'singleton' is not enforced and just allows code to abuse global objects.